Process.c

```diff
@@ -513,12 +513,16 @@ void Process_toggleTag(Process* this) {
 }
 
 bool Process_setPriority(Process* this, int priority) {
-   int old_prio = getpriority(PRIO_PROCESS, this->pid);
-   int err = setpriority(PRIO_PROCESS, this->pid, priority);
-   if (err == 0 && old_prio != getpriority(PRIO_PROCESS, this->pid)) {
-      this->nice = priority;
+   if ( Process_getuid == 0 || Process_getuid == (int) this->st_uid ) {
```

Comment:
Couldn't we just let this silently fail when you lack permissions? This is how I've always did on Linux.